test/e2e/framework/test_context.go

```diff
@@ -436,4 +438,62 @@ func AfterReadingAllFlags(t *TestContextType) {
 		}
 		os.Exit(1)
 	}
+
+	// Check for num-nodes if not set by looking for all nodes that are schedulable
+	if t.CloudConfig.NumNodes < 0 {
+		numNodes, err := countScheduleableNodes(t.KubeContext, t.KubeConfig, t.Host)
+		if err != nil {
+			klog.Errorf("Error getting num-nodes from API: %v", err)
+		} else {
+			t.CloudConfig.NumNodes = numNodes
+		}
+	}
+}
+
+// Gets a config used to create a client
+// TODO: (stevesloka) We should simplify this across the framework and not duplicate
+func loadConfig(kubeContext, kubeconfig, host string) (*restclient.Config, error) {
+	c, err := RestclientConfig(kubeContext)
+	if err != nil {
+		if kubeconfig == "" {
+			return restclient.InClusterConfig()
+		}
+		return nil, err
+	}
+	return clientcmd.NewDefaultClientConfig(*c, &clientcmd.ConfigOverrides{ClusterInfo: clientcmdapi.Cluster{Server: host}}).ClientConfig()
+}
+
+// Counts the number of schedulable nodes without "NoSchedule" taints
+func countScheduleableNodes(kubeContext, kubeconfig, host string) (int, error) {
+	// Get Config
+	config, err := loadConfig(kubeContext, kubeconfig, host)
+	if err != nil {
+		klog.Errorf("Failed to get config: %v", err)
+	}
+
+	// Get Client
+	clientSet, err := clientset.NewForConfig(config)
+	if err != nil {
+		return -1, err
+	}
+
+	// Get list of all nodes
+	nodes, err := clientSet.CoreV1().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		return -1, err
+	}
+
+	count := 0
+	for _, node := range nodes.Items {
+		if len(node.Spec.Taints) == 0 {
+			count++
+		} else {
+			for _, t := range node.Spec.Taints {
```

Comment:
If you have a clientset, why not use `isNodeSchedulable`? It seems like we should try to reuse those helper methods as much as possible.

Doesn't it include all the tweaks in logic necessary to incorporate spec settings and taints?

Reply:
Didn't know that was a thing, let me update. =)

Reply:
Actually I think `GetReadySchedulableNodesOrDie` might do both since `isNodeScheduleAble` doesn't include the taints check.